metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/ScheduledWorkflowDao.java

```diff
@@ -93,9 +96,9 @@ public ScheduledWorkflow getScheduledWorkflow(String datasetId) {
    * @return the found ScheduledWorkflow or null
    */
   public ScheduledWorkflow getScheduledWorkflowByDatasetId(String datasetId) {
-    return morphiaDatastoreProvider.getDatastore()
-        .find(ScheduledWorkflow.class).field(DATASET_ID)
-        .equal(datasetId).get();
+    return MongoRequestUtil.retryableMongoRequest(
+        () -> morphiaDatastoreProvider.getDatastore().find(ScheduledWorkflow.class)
+            .field(DATASET_ID).equal(datasetId).get());
   }
 
   /**
```

Comment:
Should the method 'exists' below not also be changed?

Reply:
I missed that, yes it should.